x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback } from 'react';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiTextArea,
+  EuiHorizontalRule,
+  EuiText,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+  EuiComboBox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import {
+  OperatingSystem,
+  ConditionEntryField,
+  isPathValid,
+  hasSimpleExecutableName,
+} from '@kbn/securitysolution-utils';
+import { isOneOfOperator } from '@kbn/securitysolution-list-utils';
+
+import { OS_TITLES } from '../../../../common/translations';
+import { ArtifactFormComponentProps } from '../../../../components/artifact_list_page';
+import {
+  CONDITIONS_HEADER,
+  CONDITIONS_HEADER_DESCRIPTION,
+  CONDITION_FIELD_DESCRIPTION,
+  CONDITION_FIELD_TITLE,
+  DESCRIPTION_LABEL,
+  DETAILS_HEADER,
+  DETAILS_HEADER_DESCRIPTION,
+  FIELD_LABEL,
+  NAME_LABEL,
+  OPERATOR_LABEL,
+  POLICY_SELECT_DESCRIPTION,
+  SELECT_OS_LABEL,
+  VALUE_LABEL,
+  ERRORS,
+} from '../../translations';
+import {
+  EffectedPolicySelect,
+  EffectedPolicySelection,
+} from '../../../../components/effected_policy_select';
+import {
+  GLOBAL_ARTIFACT_TAG,
+  BY_POLICY_ARTIFACT_TAG_PREFIX,
+} from '../../../../../../common/endpoint/service/artifacts/constants';
+import { useLicense } from '../../../../../common/hooks/use_license';
+import { isValidHash } from '../../../../../../common/endpoint/service/trusted_apps/validations';
+
+interface BlocklistEntry {
+  field: ConditionEntryField;
+  operator: 'included';
+  type: 'match_any';
+  value: string[];
+}
+
+interface ItemValidation {
+  name?: React.ReactNode[];
```

Comment:
Why are these react nodes?

I'm also having a hard time understanding the fact that they are both arrays. What is stored in them exactly?

Reply:
helpText (warnings) and errors on `EuiFormRow` are typed as React.ReactNode[]. these are the actual messages we're passing directly into `EuiFormRow`.

Reply:
Ok. thanks for that. `ReactNode` alway accepts string, but Its ok to store JSX here if that is preferable. 

Can you explain the relation ship between the array of nodes for `name` and the array of nodes for `value` in the same object? thats the part that is confusing me.

Reply:
I can probably have better naming here. the only places we need validation messages are on the `name` input and `value` input. the values under `ItemValidation.name` are all the validation messages for the `name` input and the values under `ItemValidation.value` are all the validation messages for the `value` input.